mev_inspect/provider.py

```diff
@@ -1,9 +1,19 @@
 from web3 import AsyncHTTPProvider, Web3
 
+from mev_inspect.geth_poa_middleware import geth_poa_middleware
 from mev_inspect.retry import http_retry_with_backoff_request_middleware
+from mev_inspect.utils import RPCType
 
 
-def get_base_provider(rpc: str, request_timeout: int = 500) -> Web3.AsyncHTTPProvider:
+def get_base_provider(
```

Comment:
Here, rather than checking for the RPC type that's been passed as a variable, it would be beneficial to write a function somewhere that infers the RPC type, maybe from the result of a create_block attempt?

Reply:
I agree, will commit new changes

Reply:
I believe there is a cleaner way to do this - namely the web3_clientVersion RPC Call